packages/context/README.md

```diff
@@ -0,0 +1,80 @@
+# @solid-primitives/context
+
+[![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg?style=for-the-badge)](https://lerna.js.org/)
+[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/context?style=for-the-badge&label=size)](https://bundlephobia.com/package/@solid-primitives/context)
+[![version](https://img.shields.io/npm/v/@solid-primitives/context?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/context)
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fdavedbase%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/davedbase/solid-primitives#contribution-process)
+
+Primitives simplifying or extending the SolidJS Context API.
+
+- **[`createContextProvider`](#createContextProvider)** - Create the context provider component & useContext function with types inferred from the factory function.
+
+## Installation
+
+```bash
+npm install @solid-primitives/context
+# or
+yarn add @solid-primitives/context
+```
+
+## `createContextProvider`
+
+Create the context provider component & useContext function with types inferred from the factory function.
+
+### How To use it
+
+#### Import
+
+```ts
+import { createContextProvider } from "@solid-primitives/context";
+```
+
+#### Creating the context provider and use-primitive
+
+Types of the context are inferred from it's factory function. Factory function will run when the provider component in executed. It takes the provider component `props` as it's argument, and what it returns will be available in the contexts for all the underlying components.
+
+```ts
+const [CounterProvider, useCounter] = createContextProvider((props: { initial: number }) => {
+  const [count, setCount] = createSignal(props.initial);
+  const increment = () => setCount(count() + 1);
+  return { count, increment };
+});
+```
+
+#### Place the Provider
+
+The provider component takes `props` declared as arguments of the factory functions.
+
+```tsx
+// Provide the context
+<CounterProvider initial={1}>
+  <App />
+</CounterProvider>
+```
+
+#### Use context in children components
+
+The context will by default be `T | undefined`.
+
+```ts
+// get the context
+const ctx = useCounter();
+ctx?.count(); // => 1
+```
+
+Or provide an fallback to be able to destructure safely.
+
+```ts
+const { count } = useCounter() ?? { count: () => 0 };
+```
```

Comment:
`createContext` allows providing a default value as a second argument, instead of needing to do this every time you call `useContext`. Perhaps this default could make sense as an optional second argument to `createContextProvider`?

Reply:
It takes a second parameter? I totally knew about that... 😁

Reply:
To clarify, it's an optional first parameter for `createContext`, and I'm proposing it as an optional section parameter for `createContextProvider`.

I'm trying to imagine practical use. Plausibly it'd be convenient for the default to be set by calling the factory function with no `props` argument. Or maybe the second argument should be the props argument to pass in... Not sure what's best here.

Reply:
This is actually quite a problem as the defaults shouldn't actually be needed since you should, well, provide the context if you are using it. It is definitely something to improve in the future, once somebody has the right idea, because yeah, passing an object of noop functions as default seams weird.
For now I think the simplest approach (what you've proposed actually) would be the best, as it still could be extended if needed.

Reply:
I think you should use the default and allow to overwrite it by using a value in the provider, e.g. for testing purposes.

Reply:
Don't know if it makes sense, if you use the provider the the defaults won't be used anyway, or do you want to overwrite the context values? If so, I thinks it's better to leave it up to the user to implement, instead of reserving the `value` prop name only for that. This way we keep the base `createContextProvider` as simple as possible, but there are always other primitives we could put in this package later.
For example one issue that the current `createContextProvider` has, is that the "jsx output" cannot be changed, so something like this becomes harder:
```tsx
// in the normal Provider component
return (
   <div>
      {some_additional_jsx_alongside_the_provider}
   </div>
   <Context.Provider value={value}>
      {props.children}
   </Context.Provider>
)
```